src/ApiRequestPanel.js

```diff
@@ -165,13 +164,16 @@ export class ApiRequestPanel extends AmfHelperMixin(EventsTargetMixin(HeadersPar
   }
 
   _renderServerSelector() {
-    const { amf, hideServerSelector, selectedServerType, selectedServerValue } = this;
+    const { amf, hideServerSelector, selectedServerType, selectedServerValue, serversCount, hideCustomServer } = this;
+    const hidden = (serversCount && serversCount < 1) || !!hideServerSelector;
     return html`<api-server-selector
-      .hidden=${hideServerSelector}
+      .hidden=${hidden}
```

Comment:
Sorry, this is my fault. I didn't notice that in the PR for the selector. 
This property should also have `reflect: true` configuration value. Now if you set the property you won't see any change (https://github.com/advanced-rest-client/api-server-selector/blob/stage/src/api-server-selector.js#L70).
Then in here, right now you won't see changes as `.hidden` is set to a property and not to an attribute. Preferably would be to use `?hidden="${hidden}"` for semantics.